### ISSUE TITLE
Combine multiple samples novel isoform ids

### ISSUE THREAD

**Issue by mcortes-lopez** [ablab/IsoQuant#46]
Hi, 

I ran multiple samples using a sample list, I was expecting to have similar ids for the de novo annotated isoforms, however it seems like each sample has their set of de novo isoforms. It's there any recommendation to find the common de novo isoforms across samples? 
I have tried stringtie --merge, however it produces new transcript ids without keeping the information of which isoforms were merged.

**Comment by andrewprzh** [comment]
Dear @mcortes-lopez 

I suggest to feed all the files as a single sample (e.g. list them in the file without blank lines) and provide 
`--read_group  file_name `
option. This will create a single annotation from all sources, but a the same time will do per-sample counts of genes and isoforms (including novel onces).
There are couple example in the manual https://github.com/ablab/IsoQuant#sec3.2

If you have BAM files, you can reuse them to save time.

Best
Andrey

**Comment by mcortes-lopez** [comment]
Hi @andrewprzh 
Does `--read_group` allows for 2 different options? I would like to keep the tag group but also use `file_name`.
Also, is there a way to include the new isoforms in the final count tables? 

Best, 
Mariela

**Comment by andrewprzh** [comment]
Dear @mcortes-lopez 

Unfortunately, current version does not support using two grouping strategies simultaneously. 
If you provide all files as a single samle with `--read_group tag:RG` you will still get a single GTF output and per-group counts, but not per-file counts. Current workaround is to obtain GTF with one option, and then re-run IsoQuant with another grouping. 
I'll think how this could be fixed.

New isoforms are included in the tables called `*.transcript_model_tpm.tsv` and ``*.transcript_model_counts.tsv`.
There grouped counts for novel isoforms as well.

Best
Andrey

**Comment by mcortes-lopez** [comment]
Hi @andrewprzh
I have been trying to run it using the suggested GTF file. I am using the `*.extended_annotation.gtf`, as I want to also include the annotation used in the first run, however the file seems to be mis-formatted with `gffutils` trowing an error of `non unique features.id`. Do you have any suggestion on the cleaning up of that annotation?

**Comment by andrewprzh** [comment]
Hi @mcortes-lopez 

That sound like a bug. Is there a possiblity to send this GTF over to me, e.g. via email or Google drive?
This would help me to figure out the problem.

Best
Andrey

**Comment by mcortes-lopez** [comment]
Hi Andrew, 
Sorry for the delay, here are the example files: https://drive.google.com/file/d/1ti43Xd1qtxHGgp29PRKf5jgYEAAUCLzx/view?usp=share_link
I solved the issue parsing the original file like this: 

```
sed -e 's/\(.*transcript_id "ENS.*Canonical \).*Canonical.*/\1"True";/g' SF3B1_combined.extended_annotation.gtf | \
 sed -e 's/\(.*transcript_id "ENST.*Canonical \).*/\1"True";/g' | awk -F'\t' '!seen[$9]++' | \
 sed -e 's/\(.*transcript_id "ENS.*".*exons "[0-9]*";\).*/\1 Canonical "True"/g'| \
 awk -F'\t' '!seen[$9]++'  > SF3B1_combined.extended_annotation_clean.gtf
```

The annotations have the following issues: 

- Transcripts which have 2 times the tag "Canonical X";"
- Transcripts which appear in two versions, a "Canonical "True";" and a "Canonical "False";" which migth change in a few nucleotides at the end, I chose for keeping the first annotated one.
- Transcripts which do not have the "Canonical" label.

**Comment by andrewprzh** [comment]
Hi @mcortes-lopez  

Thanks for sharing the data! 

Meanwhile, I already fixed the problem with duplicate entries in extended GTF. Will do a new release soon.
I will take a look at the problem with the canonical attribute ASAP.

Best
Andrey

**Comment by andrewprzh** [comment]
Hi @mcortes-lopez 

Found the bug, will be fixed in the next release, thanks a lot for noticing!

Best
Andrey

**Comment by andrewprzh** [comment]
Dear @mcortes-lopez 

Duplicated transcript  and "Canonical" attribute issues are now fixed in the recent release: https://github.com/ablab/IsoQuant/releases/tag/v3.1.2

Best
Andrey

**Comment by mcortes-lopez** [comment]
Hi @andrewprzh,
Thanks for the update! I was testing it this morning and it seems like the duplicated "Canonical" attribute is gone but the duplicated transcripts are still in there. 

Best, 
Mariela

**Comment by andrewprzh** [comment]
Hi @mcortes-lopez 

Hm, that's unpleasant if 3.1.2 still has this bug, could you send me the command line and the GTF (or a part of it with duplicated transcripts)? 

Best
Andrey

**Comment by mcortes-lopez** [comment]
I have added my gtf files (original and cleaned - using a similar approach to the one I described before-) in this [link](https://drive.google.com/file/d/1ti43Xd1qtxHGgp29PRKf5jgYEAAUCLzx/view?usp=share_link) as above. I got the warning on the first of these duplicated transcripts: "ENST00000466430" and just doing `awk 'seen[$0]'` over the original file I saw a lot of duplicated lines.

**Comment by andrewprzh** [comment]
The files you've sent are generated with IsoQuant 3.0.3, but does this issue reproduces with the latest 3.1.2 version (released just a few hours ago)?

Best
Andrey

**Comment by andrewprzh** [comment]
@mcortes-lopez  
Thanks again for quick reports!

Sorry for the inconvenience, hopefully, this is the final fix for this issue :) 
Double checked on my data and uploaded a fixed version of 3.1.2 [here](https://github.com/ablab/IsoQuant/releases/tag/v3.1.2), will take some time to appear on bioconda.

Best
Andrey
